### PR TITLE
Support for local csv file as input

### PR DESCRIPTION
**Pull request by CannedFish**
In some scenarios, such as parameter tuning, it is not necessary to obtain relevant statistics through the network every time. This PR help to support batch reading statistics from local csv files.

Usage as follows:
```
usage: criticality_score [-h] (--repo REPO | --local-file L_FILE) [--format {default,csv,json}]
                         [--params PARAMS [PARAMS ...]]

Gives criticality score for an open source project

optional arguments:
  -h, --help            show this help message and exit
  --repo REPO           repository url
  --local-file L_FILE   path of a local csv file with repo stats
  --format {default,csv,json}
                        output format. allowed values are [default, csv, json]
  --params PARAMS [PARAMS ...]
                        Additional parameters in form (<value>|<key>):<weight>:<max_threshold>, <value> for repo, and
                        <key> for loacl file
```

Here is an example:
```
# cat python_top_2.csv
name,artifacts,url,language,created_since,updated_since,contributor_count,org_count,commit_frequency,recent_releases_count,updated_issues_count,closed_issues_count,comment_frequency,dependents_count,criticality_score,idc,bc,cc
salt,salt@3004,https://github.com/saltstack/salt,Python,120,0,3636,7,62.6,21,1641,679,1.1,20974,0.87814,6.436596945448553e-05,3.4932693324662978e-06,1.0844059851056442e-19
core,homeassistant@2022.2.9,https://github.com/home-assistant/core,Python,89,0,2520,9,162.7,205,5330,3786,3.5,380,0.87147,1.2873193890897107e-05,6.431467681977665e-06,0.12111046542342438

# criticality_score --local-file python_top_2.csv --format csv --params idc:1.5:0.1 bc:1.5:0.01 cc:0.5:1
name,artifacts,url,language,created_since,updated_since,contributor_count,org_count,commit_frequency,recent_releases_count,updated_issues_count,closed_issues_count,comment_frequency,dependents_count,criticality_score,idc,bc,cc
salt,salt@3004,https://github.com/saltstack/salt,Python,120,0,3636,7,62.6,21,1641,679,1.1,20974,0.62214,6.436596945448553e-05,3.4932693324662978e-06,1.0844059851056442e-19
core,homeassistant@2022.2.9,https://github.com/home-assistant/core,Python,89,0,2520,9,162.7,205,5330,3786,3.5,380,0.62426,1.2873193890897107e-05,6.431467681977665e-06,0.12111046542342438
```

### CODE REVIEW

**Review by calebbrown — CHANGES_REQUESTED** [review]
Thanks for this change. I ripped the score logic out into a separate file and did the csv processing in there, but this helps keep it all together.


FYI: See https://github.com/ossf/criticality_score/issues/108 on the future direction I have in mind for this project.

File: criticality_score/run.py [hunk 1]
```diff
@@ -431,20 +431,17 @@ def get_param_score(param, max_value, weight=1):
     return (math.log(1 + param) / math.log(1 + max(param, max_value))) * weight
 
 
-def get_repository_stats(repo, additional_params=None):
+def get_repository_stats(repo, additional_params=None, result_dict=None):
```
[the remark this answers is not in the document]
Comment (CannedFish) [reply criticality_score/run.py]: Hi @calebbrown ，thanks for your reply. 

I have simplified the main function by handling repo and local csv file in different functions. Please review again.

By the way, as you said in #108 , is that mean this PR will not be accepted until the end of April at the earliest?

File: criticality_score/run.py [hunk 2]
```diff
@@ -639,37 +680,13 @@ def main():
 
     output = None
     if args.repo:
```
Comment (calebbrown) [inline criticality_score/run.py:682]: I suggest adding an `else` condition here to handle the situation were neither `args.repo` nor `args.l_file` are present.

**Review by calebbrown — COMMENTED** [review]
This looks great!

I have one minor comment, but otherwise this is ready to go.

File: criticality_score/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (calebbrown) [reply criticality_score/run.py]: Thanks! I'm happy for outstanding PRs to be reviewed and merged.

For what it's worth, PRs like this are useful signal for features/capabilities any rewrite needs to consider.

File: criticality_score/run.py (hunk 2 above)
Comment (CannedFish) [reply criticality_score/run.py:682]: Hi @calebbrown ，I've added an else to catch unknown options. I thought that the required option of the parser group was already set to true, so I didn't add an else. But you're right, it's better to add an else here.

**Comment by CannedFish** [comment]
Hi @calebbrown , I have resolved confilcts, please review again. Thanks.

**Review by calebbrown — APPROVED** [review]

**Pull request merged**